samples/15.e.clear-conversation/README.md

```diff
@@ -0,0 +1,297 @@
++            timer.start();
++        }
+        return next(action);
+    }
+);
+```
+
+Next, we are going to create a `deleteConversation` function that the timer will call when the session expires. The function will iterate through and delete all of the activities from the store. Then it will disconnect from DirectLine and request a new token to start a new conversation.
+
+```diff
++ async function deleteConversation() {
++     const { activities } = store.getState();
++ 
++     // Delete Activities 
++     activities.forEach( ({ id }) => {
++         store.dispatch({
++             type: 'DIRECT_LINE/DELETE_ACTIVITY',
++             payload: {
++                 activityID: id
++             }
++         });
++     });
++ 
++     // Disconnect DirectLine
++     store.dispatch({
++         type: 'DIRECT_LINE/DISCONNECT'
++     });
++ 
++     // Connect to a new DirectLine conversation
++     const res = await fetch('https://thdurn-all-channels-directline-token.azurewebsites.net/+ directline/token', { method: 'POST' });
++     const { token } = await res.json();
++ 
++     store.dispatch({
++         type: 'DIRECT_LINE/CONNECT',
++         payload: {
++             directLine: window.WebChat.createDirectLine({ token }),
++         }
++     });
++ }
+```
+
+Finally, pass in your new store to the Web Chat render method, and that's it.
+
+```diff
+window.WebChat.renderWebChat({
+  directLine: window.WebChat.createDirectLine({ token }),
++ store
+}, document.getElementById('webchat'));
+```
+
+## Completed code
+
+Here is the finished `index.html`:
+
+```diff
+<!DOCTYPE html>
+<html lang="en-US">
+  <head>
+-    <title>Web Chat: Full-featured bundle</title>
++    <title>Web Chat: Clear chat history</title>
+    <!--
+      For demonstration purposes, we are using the development branch of Web Chat at "/master/webchat.js".
+      When you are using Web Chat for production, you should use the latest stable release at "/latest/webchat.js",
+      or lock down on a specific version with the following format: "/4.1.0/webchat.js".
+    -->
+    <script src="https://cdn.botframework.com/botframework-webchat/master/webchat.js"></script>
+    <style>
+        html, body { height: 100% }
+-       body { margin: 0 }
++       body { 
++           display: flex;
++           flex-direction: column;
++           height: 100%;
++           margin: 0 
++       }
++
+        #webchat {
++           flex: 1;
+            width: 100%;
+        }
+
++       div.time-remaining {
++           padding: 10px 10px; 
++
++           color: #807f7f;
++           text-align: right;
++           font-family: "Calibri", "Helvetica Neue", Arial, sans-serif;
++       }
+    </style>
+  </head>
+  <body>
++   <div class="time-remaining">
++       Time Remaining: <b><span id="timer">0:00</span></b>
++   </div>
+    <div id="webchat" role="main"></div>
+    
+    <script>
+
+      (async function () {
+        // In this demo, we are using Direct Line token from MockBot.
```

Comment:
Remove all repetitive comments in 'finished' code.

Reply:
Generally I try to remove all comments in the finished code to make it more polished. If they want to look at the comments, they can always look at the `index.html`